### PR TITLE
Converted to typescript

### PR DESCRIPTION
**Pull request by mikehazell**
- typescript
- prettier
- added some basic editor niceness in .editorconfig
- git-ignored the most basic basics
- specified the `files` to be included in the npm package.

### CODE REVIEW

File: lib/manylicenses.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env node
```
Comment (mikehazell) [inline lib/manylicenses.js:1]: Including the compiled code in the repo.. I don't have strong opinions on this.

File: package.json [hunk 2]
```diff
@@ -4,13 +4,26 @@
   "license": "MIT",
   "type": "module",
   "author": "Thinkmill Labs Pty Ltd",
-  "repository": "https://github.com/thinkmill/manylicenses",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/thinkmill/manylicenses.git"
+  },
```
Comment (mikehazell) [inline package.json:10]: npm asks for this thusly:
https://docs.npmjs.com/cli/v7/configuring-npm/package-json#repository

File: package.json [hunk 3]
```diff
@@ -4,13 +4,26 @@
...
+  "files": [
+    "lib/*"
+  ],
```
Comment (mikehazell) [inline package.json:13]: Only include `lib` in the package, plus any files npm includes by default like package.json, LICENSE etc.

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+"@types/node@^15.6.1":
+  version "15.6.1"
+  resolved "https://registry.yarnpkg.com/@types/node/-/node-15.6.1.tgz#32d43390d5c62c5b6ec486a9bc9c59544de39a08"
+  integrity sha512-7EIraBEyRHEe7CH+Fm1XvgqU6uwZN8Q7jppJGcqjROMT29qhAuuOxYB1uEY5UMYQKEmA5D+5tBnhdaPXSsLONA==
+
+prettier@^2.3.0:
+  version "2.3.0"
+  resolved "https://registry.yarnpkg.com/prettier/-/prettier-2.3.0.tgz#b6a5bf1284026ae640f17f7ff5658a7567fc0d18"
+  integrity sha512-kXtO4s0Lz/DW/IJ9QdWhAf7/NmPWQXkFr/r/WkR3vyI+0v8amTDxiaQSLzs8NBlytfLWX/7uQUMIW677yLKl4w==
```
Comment (dcousens) [inline yarn.lock:13]: Wait, `prettier` has no transitives?

File: src/manylicenses.ts [hunk 5]
```diff
@@ -53,17 +53,27 @@ Options:
...
+      excludePrefix?: string | string[]
+    }
+  | undefined
+
 try {
-  ({ manylicenses } = JSON.parse(fs.readFileSync(`${process.cwd()}/package.json`)))
+  const pkg = JSON.parse(
+    fs.readFileSync(`${process.cwd()}/package.json`).toString()
+  )
+  manylicenses = pkg.manylicenses
```
Comment (mikehazell) [inline src/manylicenses.ts:73]: Prettier really wanted to put a semicolon before this line.

File: src/manylicenses.ts [hunk 6]
```diff
@@ -117,9 +133,12 @@ for (const rowArray of body) {
...
   // skip if prefix matches
-  if (excludePrefixes.some((prefix) => {
-    return name.startsWith(prefix)
-  })) continue
+  if (
+    excludePrefixes.some((prefix) => {
+      return name.startsWith(prefix)
+    })
+  )
+    continue
```
Comment (dcousens) [inline src/manylicenses.ts:141]:
```suggestion
    })
  ) {
    continue
  }
```

**Review by dcousens — APPROVED** [review]
LGTM,  nice addition! :+1:

**Pull request merged**